Roles/Impostor/Exorcist.cs

```diff
@@ -0,0 +1,254 @@
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static bool IsExorcismCurrentlyActive()
+    {
+        return IsExorcismActive;
+    }
+
+    public static void ExorcisePlayer(PlayerControl player)
+    {
+        if (ExorcismEndOnKill.GetBool() && IsExorcismActive)
+        {
+            IsExorcismActive = false;
+            Utils.SendMessage(Translator.GetString("ExorcistEnd"));
+        }
+        player.SetDeathReason(PlayerState.DeathReason.Exorcised);
+        player.SetRealKiller(ExorcistPlayer);
+        GuessManager.RpcGuesserMurderPlayer(player);
+        Main.PlayersDiedInMeeting.Add(player.PlayerId);
+        MurderPlayerPatch.AfterPlayerDeathTasks(player, PlayerControl.LocalPlayer, true);
+        Utils.SendMessage(string.Format(Translator.GetString("ExorcistKill"), player.name.RemoveHtmlTags()));
+        Exorcist exorcist = (Exorcist)ExorcistPlayer.GetRoleClass();
+        exorcist.Sacrifice();
+    }
+
+    public static void ActivateExorcism(PlayerControl player)
+    {
+        var exorcist = (Exorcist)player.GetRoleClass();
+        exorcist.ExorcismLimitPerMeeting--;
+        exorcist.AbilityLimit--;
+        exorcist.SendSkillRPC();
+
+        if (TryHideMsg.GetBool())
+            GuessManager.TryHideMsg();
+        ExorcistPlayer = player;
+        IsDelayActive = true;
+        if (ExorcismDelay.GetFloat() > 0)
+            Utils.SendMessage(string.Format(GetString("ExorcistNotify"), ExorcismDelay.GetFloat()));
+
+        _ = new LateTask(() =>
+        {
+            IsExorcismActive = true;
+            IsDelayActive = false;
+            Utils.SendMessage(string.Format(Translator.GetString("ExorcistStart"), ExorcismActiveFor.GetFloat()));
+            _ = new LateTask(() =>
+            {
+                if (IsExorcismActive)
+                {
+                    IsExorcismActive = false;
+                    Utils.SendMessage(GetString("ExorcistEnd"));
+                }
+            }, ExorcismActiveFor.GetFloat(), "ExorcistNotify");
+
+        }, ExorcismDelay.GetFloat(), "ExorcistNotify");
+    }
+
+    public void Sacrifice()
+    {
+        Sacrifices++;
+        if (Sacrifices >= ExorcismSacrificesToDispel.GetInt())
+            Dispelled = true;
+    }
+    public override string GetProgressText(byte playerId, bool coooms)
+        => Utils.ColorString(AbilityLimit <= 0 ? Color.gray : Utils.GetRoleColor(CustomRoles.Exorcist), $"({AbilityLimit})") ?? "Invalid";
+
+    [HarmonyPatch(typeof(MeetingHud), nameof(MeetingHud.Start))]
+    class StartMeetingPatch
+    {
+        public static void Postfix(MeetingHud __instance)
+        {
+            if (PlayerControl.LocalPlayer.Is(CustomRoles.Exorcist) && PlayerControl.LocalPlayer.IsAlive())
+                CreateExorcistButton(__instance);
+        }
+    }
+
+    public static void CreateExorcistButton(MeetingHud __instance)
+    {
+        PlayerControl pc = PlayerControl.LocalPlayer;
```

Comment:
check swapper code
no matter player is alive or not
first destory the existing button

if player is alive, then spawn it